### ISSUE TITLE
Uncaught TypeError: Cannot set property 'localAddress' of undefined

### ISSUE THREAD

**Issue by 50Wliu** [atom/atom#5468]
Steps to reproduce:
1. Nothing?  I left my laptop on for a bit and it eventually went into sleep mode while I did other stuff.  Interesting information is at the bottom of this report.

**Atom Version**: 0.177.0
**System**: linux 3.13.0-43-generic (Ubuntu 14.10)
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot set property 'localAddress' of undefined

```
At net.js:840

TypeError: Cannot set property 'localAddress' of undefined
  at connect (net.js:840:23)
  at net.js:943:9
  at dns.js:87:18
  at process._tickCallback (node.js:364:11)

```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "archive-view",
      "autoflow",
      "autosave",
      "background-tips",
      "deprecation-cop",
      "encoding-selector",
      "image-view",
      "incompatible-packages",
      "open-on-github",
      "release-notes",
      "spell-check",
      "styleguide",
      "welcome",
      "wrap-guide",
      "package-generator"
    ],
    "themes": [
      "isotope-light-ui",
      "fizzy"
    ]
  },
  "editor": {
    "fontFamily": "Inconsolata",
    "fontSize": 14,
    "showIndentGuide": true,
    "tabLength": 4,
    "softTabs": false,
    "invisibles": {}
  }
}
```

### Installed Packages

```coffee
# User
file-icons, v1.4.9
fizzy, v0.15.0
isotope-light-ui, v1.0.0
jenkins, v0.3.1
language-sourcepawn, v0.4.1
linter, v0.11.1
linter-jshint, v0.1.0
sourcepawn-compiler, v0.0.0
travis-ci-status, v0.11.2

# Dev
No dev packages
```

Some interesting things to note:
1. This error got thrown multiple times-Notifications is showing 5 errors, all identical.
2. The first error got formatted super-weirdly (see the picture below).
3. Clicking on "Create issue on atom/atom" does...nothing :(.  Had to copy the issue to clipboard and paste it here.
4. This error may be due to the Jenkins package, but I wanted to submit it here first just to make sure.

Picture:
![screenshot from 2015-02-09 14 15 33](https://cloud.githubusercontent.com/assets/2766036/6113670/181784c6-b067-11e4-9570-aeaad8220ab0.png)

**Comment by 50Wliu** [comment]
Ooh, figured out why the automatic issue creator wasn't working.
![screenshot from 2015-02-09 14 22 14](https://cloud.githubusercontent.com/assets/2766036/6113677/2006a8a6-b067-11e4-8a4d-4bf9085b4829.png)
The full URL is https://api.github.com/search/issues?q=Uncaught%20TypeError:%20Cannot%20set%20property%20%27localAddress%27%20of%20undefined%20repo:atom/atom&sort=created.

**Comment by kevinsawicki** [comment]
Looks similar to https://github.com/atom/atom/issues/5389

**Comment by 50Wliu** [comment]
Yeah, but the reproduction steps are way different (in addition, I don't have Beautify, but perhaps this can be caused by different packages?).

**Comment by kevinsawicki** [comment]
I wonder if it is https://github.com/iojs/io.js/issues/510 and fixed already upstream in io.js

**Comment by 50Wliu** [comment]
Any updates on when io.js will be updated in Atom to include the linked fix?  I'm still getting this in 0.180.0.

**Comment by topmotiveret** [comment]
Get the same/simllar error on OS X 10.10.2 

n
![screen shot 2015-02-24 at 11 23 07](https://cloud.githubusercontent.com/assets/7926595/6347350/9b5d9288-bc17-11e4-855c-691d66ef253a.png)

**Comment by jekriske** [comment]
Still a problem in Atom 0.187.0

**Comment by wilrnh** [comment]
Yep, seems to happen pretty much everytime I go to sleep. Have only been using Atom for 2 days now though.

**Comment by daithi-coombes** [comment]
I can confirm this bug:
atom: version 0.187.0

Is it possible to update `io.js` outside of the `atom` ide, if it is a bug fixed in io.js?
```
TypeError: Cannot set property 'localAddress' of undefined
  at connect (net.js:840:23)
  at net.js:943:9
  at GetAddrInfoReqWrap.asyncCallback [as callback] (dns.js:83:16)
  at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:101:10)
```

**Comment by kevinsawicki** [comment]
This should be fixed by https://github.com/atom/notifications/pull/62 and available in Atom 1.0.0+
